src/main/java/de/tum/in/www1/artemis/service/util/structureoraclegenerator/ClassesDiff.java

```diff
@@ -0,0 +1,124 @@
+        attributesDiff.removeIf(fieldIsImplicit);
+
+        // If the template is non-existent, then the attributes diff consists of all the attributes of the solution type.
+        if(templateClass != null) {
+
+            // Check all the attributes in the template type if they match to the ones in the solution type
+            // and remove them from the diff, if that's the case.
+            for (CtField<?> templateAttribute : templateClass.getFields()) {
+
+                // The fields are uniquely identified by their names.
+                attributesDiff.removeIf(solutionAttribute ->
+                    solutionAttribute.getSimpleName().equals(templateAttribute.getSimpleName()));
+            }
+        }
+
+        return attributesDiff;
+    }
+
+    /**
+     * This method generates the constructors diff of the solution and template type, e.g. the constructors defined in the
+     * solution type but not in the template type.
+     * @return A set of constructors defined in the solution type but not in the template type.
+     */
+    private Set<CtConstructor<?>> generateConstructorsDiff() {
+        // Use this predicate to filter out methods that are implicit, e.g. not explicitly defined in the code.
+        Predicate<CtConstructor<?>> constructorIsImplicit = CtElement::isImplicit;
+
+        // Create an empty set of constructors for the constructors diff and deep-copy the constructors of the solution type in it.
+        Set<CtConstructor<?>> constructorsDiff = new HashSet<>(solutionClass.getConstructors());
+        constructorsDiff.removeIf(constructorIsImplicit);
+
+        // If the template is non-existent, then the constructors diff consists of all the constructors of the solution type.
+        if(templateClass != null) {
+
+            // Check all the constructors in the template type if they match to the ones in the solution type
+            // and remove them from the diff, if that's the case.
+            for(CtConstructor<?> templateConstructor : templateClass.getConstructors()) {
+
+                // The constructors are uniquely identified by their parameter types.
+                constructorsDiff.removeIf(solutionConstructor -> parameterTypesAreEqual(solutionConstructor, templateConstructor));
+            }
+
+        }
+
+        return constructorsDiff;
+    }
+
+    /**
+     * This method checks if the parameter types of an executable in the solution type are the same to an executable
+     * in the template type. An executable can be a method or a constructor.
+     * @param solutionExecutable: The executable present in the solution type.
+     * @param templateExecutable: The executable present in the template type.
+     * @return True, if the parameter types are the same, false otherwise.
+     */
+    protected boolean parameterTypesAreEqual(CtExecutable<?> solutionExecutable, CtExecutable<?> templateExecutable) {
```

Comment:
This code seems to be duplicated from TypesDiff. Can we deduplicate it or is this not possible?

Reply:
I did that